WordPress/gutenberg
# Global Styles: Update `gutenberg_get_global_stylesheet` to use `WP_Object_Cache`

Part of https://github.com/WordPress/gutenberg/issues/45171
Fixes https://github.com/WordPress/gutenberg/issues/43914

## What?

This PR updates `gutenberg_get_global_stylesheet`, so the runtime cache is `WP_Object_Cache` and not a transient. Furthermore, the stylesheet is cached indefinitely now (previously it was cached for a minute) and invalidated once it's needed to regenerate it (e.g. after the Global Styles CPT changes, after a theme switch, after a theme update).


## Why?

See https://core.trac.wordpress.org/ticket/56910:

> These caches should be set forever and correctly invalidated. This will improve performance and mean that front end users who hit a page will not suffer poor performance.

## How?
- Replace `transient_` functions with `wp_cache_` functions.
- Hook into the relevant actions to invalidate the cache.

## Testing Instructions

* Use a block theme, such as TT3 or TT2
* Go to the site editor
* Switch to a different style variation (Click global styles > Browse styles > Pick one). Save the changes.
* Click view > View site. 
* Make sure the new styles are applied immediately.
